node/subsystem/src/messages.rs

```diff
@@ -240,19 +257,28 @@ pub enum AvailabilityStoreMessage {
 
 	/// Query whether a `AvailableData` exists within the AV Store.
 	///
-	/// This is useful in cases like bitfield signing, when existence
+	/// This is useful in cases when existence
 	/// matters, but we don't want to necessarily pass around multiple
 	/// megabytes of data to get a single bit of information.
 	QueryDataAvailability(Hash, oneshot::Sender<bool>),
 
 	/// Query an `ErasureChunk` from the AV store.
 	QueryChunk(Hash, ValidatorIndex, oneshot::Sender<Option<ErasureChunk>>),
 
+	/// Query whether an `ErasureChunk` exists within the AV Store.
+	///
+	/// This is useful in cases like bitfield signing, when existence
+	/// matters, but we don't want to necessarily pass around large
+	/// quantities of data to get a single bit of information.
+	QueryChunkAvailability(Hash, ValidatorIndex, oneshot::Sender<bool>),
```

Comment:
I had serious doubts as I was adding `QueryDataAvaialbility` message; is it really better for performance? We anyway lift one big blob from disk, there is no way to ask `kvdb` for just availability; when we send whatever data struct that's containing the pointer to this blob, only it will be sent around, wouldn't it?

Reply:
At worst, it masks a `kvdb` limitation which can be fixed in the future.

Reply:
https://github.com/paritytech/parity-common/pull/409